core/trino-main/src/main/java/io/trino/operator/HttpPageBufferClient.java

```diff
@@ -485,6 +500,13 @@ public void onFailure(Throwable t)
         }, pageBufferClientCallbackExecutor);
     }
 
+    @VisibleForTesting
+    synchronized void updateAverageRequestSize(int successfulRequests, long responseSize)
+    {
+        // AVG_n = AVG_(n-1) * (n-1)/n + VALUE_n / n
+        averageRequestSizeInBytes = (long) (1.0 * averageRequestSizeInBytes * (successfulRequests - 1)) / successfulRequests + responseSize / successfulRequests;
```

Comment:
nit: as a follow-up?
```
if (responseSize == 0) {
  averageRequestSizeInBytes = 0;
  requestsSucceeded.set(0); // NOGO because you have to have a separate counter
}
```

Reply:
ok